pilot/pkg/model/jwks_resolver.go

```diff
@@ -61,6 +61,11 @@ const (
 	// JwtPubKeyRefreshIntervalOnFailureResetThreshold is the threshold to reset the refresh interval on failure.
 	JwtPubKeyRefreshIntervalOnFailureResetThreshold = 60 * time.Minute
 
+	// How many times should we attempt to update a cache bucket via load + compare and swap before giving up.
+	// Realistically, there would rarely be more than 2 goroutines competing to write to any cache bucket at any given time.
```

Comment:
I think from the convertToEnvoyJwtConfig flow -> GetPublicKey there can be a lot more than 2 fwiw

Reply:
Are multiple policy configs able to be built in parallel using the same issuers behind the scenes? I'm new to the codebase so may have misinterpreted how this is being used. 

Trying again in these compare and swap iterations is extremely so we should try as many times as we expect to have competitors trying to update a given cache bucket, since one competitor is always guaranteed to succeed in this pattern. I can edit this comment to clarify since it's not true.

Reply:
I landed on this approach because I felt the trade-offs made here were reasonable. In the case refresh somehow fails to do this it'll forcefully refresh the cache. I didn't want to have to manage a per-bucket mutex since it would complicate things somewhat.

Reply:
Unfortunately yes. It traces back to https://github.com/istio/istio/blob/33fb7a36bd52e40f7871fc5e6c1ae5ea26bf3230/pilot/pkg/xds/lds.go#L104 which is _generally_ controlled concurrency based on https://github.com/istio/istio/blob/a57ded2280890f4bdd95363ea0b23ee5de30866b/pilot/pkg/xds/discovery.go#L464 which defaults to 20-100 (based on CPU size). There are some less common codepaths that can trigger it too though, so don't rely on any hard limit.

I think the approach is fine, just the comment maybe misleading

Reply:
Ok good to know. If there is a high degree of contention in writes then chances are forcefully overwriting the cache from the refresher will be fine since it’ll grab a recent last used time that was set by one of the many GetPublicKey calls. So I think we’re ok with a low concurrency attempt limit. I will amend this comment for clarity.

Reply:
Ok I think my latest iteration includes more accurate explanations of the write-contention scenario. Please check my logic on this latest push.